src/book/02-system/03-tools/05-system-configuration.mdx

```diff
@@ -0,0 +1,83 @@
+---
+section: System
+chapter: Tools
+title: System Configuration
+description: A tool that sets up the robot.
+slug: /system/tools/system_configuration
+---
+
+## Description
+
+This module is used to ensure that the configuration that is running on a device matches what is required.
+
+It goes through several steps to ensure the system is set up correctly.
+
+### Filesystem
+
+The system will look through each file in the `module/tools/SystemConfiguration/data/system/default` directory.
+For each of these files it will see if there is an equivalent file in the
+`module/tools/SystemConfiguration/data/system/$HOSTNAME` directory and use that, otherwise it uses the default file.
+Using this file it will check that the same file in the root directory is identical to this one, and if not it will copy it across.
+This will ensure that all the system configuration files match the files that are in the repository.
+
+<Alert type='info'>
+The system will only look at files that exist in the `module/tools/SystemConfiguration/data/system/default` directory.
+If you have a file that will override for a specific system, you must provide a default file as well.
+After this, it will go through the list of permissions changes that are in the `SystemConfiguration.yaml` file and apply them.
+This configuration file must list the total permissions as an octal number.
+</Alert>
+<Alert type='info'>
+SystemConfiguration will not delete files that are not in `system/default`,
+even if they have been previously installed by SystemConfiguration.
+</Alert>
+
+After permissions are set, it will go through the list of symlinks that are in the `SystemConfiguration.yaml` and ensure they are created.
+Checks are made to ensure that the link target exists, and the link doesn't exist.
+
+- If the target doesn't exist, no link is made
+- If the link name already exists and is a regular file, the file is renamed to `<link name>.old` and the link is made
+- If the link name already exists and **is not** a symlink to the target, the link is removed and the correct link is made
+- If the link name already exists and **is** a symlink to the target, nothing is done
+- If the link name doesn't exist, the link is made
+
+### Packages
+
+After this the system will then attempt to update and upgrade the system using the package manager.
```

Comment:
Maybe mention that the system configuration loads the mirror file that's used in the docker container, so it'll upgrade/downgrade to whatever the date locked version was in the docker image you installed from